daemons/attrd/attrd_corosync.c

```diff
@@ -230,6 +272,7 @@ attrd_peer_change_cb(enum crm_status_type kind, crm_node_t *peer, const void *da
     // Remove votes from cluster nodes that leave, in case election in progress
     if (gone && !is_remote) {
         attrd_remove_voter(peer);
+        attrd_remove_peer_protocol_ver(peer->uname);
```

Comment:
If a peer leaves, we also have to remove it from respondents for all pending confirmations (and it's just as good as getting a confirmation)

We also need to stop waiting on confirmations if the client drops while waiting

I don't think we need to worry about newly joined peers, since they will sync their attributes with the rest of the cluster, and that sync request should be processed after the update.

Reply:
Added code to handle both of these possibilities, though I'm not sure I know of a way to test them.